### PR TITLE
Safer code

### PR DESCRIPTION
**Pull request by lex148**
The goal of this branch is to:
- remove all unwrap calls ( If they can't be removed, convert to expect ) 
- remove unsafe accessing of Vec [index] convert to .get(index) so rust will bounds check

### CODE REVIEW

File: src/utils/helpers.rs [hunk 1]
```diff
@@ -59,18 +53,16 @@ where
...
     let len = list.len() as i32;
-    let (index, item) = match list.iter().enumerate().find(|&x| test(x.1)) {
-        Some(x) => (x.0, x.1.clone()),
-        None => {
-            return;
-        }
-    };
+
+    let index = list.iter().enumerate().find(|&x| test(x.1))?.0;
+    let item = list.get(0)?.clone();
```
Comment (AethanFoot) [inline src/utils/helpers.rs:64]: doesn't this need to be index not 0?

**Review by AethanFoot — COMMENTED** [review]
This is great, most definitely taking some mental notes on this.

File: src/utils/helpers.rs (hunk 1 above)
Comment (lex148) [reply src/utils/helpers.rs]: good catch thanks!

**Comment by lex148** [comment]
Ok, This is ready for others to look at if you want. 
A couple notes about this PR:
- I didn't touch unwraps in tests. This seams appropriate. If a test panics you most likely want to know
- My intent was to not change functionally anywhere with the one exception of the mutability cleanup in ExternalCommandHandler

**Pull request merged**